### DISCUSSION TITLE
ImageDataGenerator without classes

### DISCUSSION THREAD

**Discussion by ZephyrusBlaze** [community/community#107548] [Programming Help] [answered]
### Body

How can I use the ImageDataGenerator in Keras to handle images without organizing them into separate classes within folders? For example, if I have a collection of images for training that aren't categorized into separate folders for classes.

I want to utilize the grayscale version of the respective image as the input (X) and its RGB counterpart as the output (y).

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by faridnec** [comment, marked as answer]
Hey, You can perhaps use the `flow_from_directory` method of `ImageDataGenerator` in Keras to handle images even if they are not organized into separate classes within folders. Instead of organizing images into separate class folders, you can organize them all into a single directory and specify the class_mode parameter as None. This tells Keras not to expect subdirectories for classes.

```python
train_generator = datagen.flow_from_directory(
        train_dir,
        target_size=(image_height, image_width),
        batch_size=batch_size,
        class_mode=None,  # Set class_mode to None since there are no classes
...................................................
```
